lib/src/presentation/views/now_playing_movies_pageview.dart

```diff
@@ -0,0 +1,93 @@
+import 'package:flutter/material.dart';
+
+import '../../core/parameter/status.dart';
+import '../../core/util/ui_constants.dart';
+import '../../domain/entity/movie_event.dart';
+import '../bloc/i_now_playing_bloc.dart';
+import '../bloc/now_playing_bloc.dart';
+import '../widgets/app_drawer.dart';
+import '../widgets/custom_progress_indicator.dart';
+import '../widgets/page_view_item.dart';
+import '../widgets/plain_text.dart';
+
+class NowPlayingMoviesPageView extends StatefulWidget {
+  NowPlayingMoviesPageView({
+    super.key,
+    required this.nowPlayingBloc,
+  });
+
+  final INowPlayingBloc nowPlayingBloc;
+  static const String routeName = '/nowPlayingMoviesPageView';
+  static const String emptyMsg =
+      'Sorry, it seems that we don´t have any movies in this category to show you!';
+  static const String title = 'Now playing movies';
+
+  @override
+  State<NowPlayingMoviesPageView> createState() =>
+      _NowPlayingMoviesPageViewState();
+}
+
+class _NowPlayingMoviesPageViewState extends State<NowPlayingMoviesPageView> {
+  @override
+  void initState() {
+    super.initState();
+    widget.nowPlayingBloc.getNowPlaying();
+  }
+
+  @override
+  void dispose() {
+    widget.nowPlayingBloc.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final PageController controller = PageController();
+    return SafeArea(
+      child: Scaffold(
+        appBar: AppBar(
+          backgroundColor: UIConstants.backgroundColor,
+        ),
+        drawer: const AppDrawer(),
+        backgroundColor: UIConstants.backgroundColor,
+        body: StreamBuilder<MovieEvent>(
+          initialData: MovieEvent(
+            status: Status.loading,
+          ),
+          stream: widget.nowPlayingBloc.moviesStream,
+          builder: (BuildContext context, AsyncSnapshot<MovieEvent> snapshot) {
+            Status status = snapshot.data!.status;
+            switch (status) {
+              case Status.loading:
+                return const CustomProgressIndicator();
+              case Status.success:
+                return PageView(
+                  controller: controller,
+                  children: snapshot.data!.moviesList!
+                      .map(
+                        (movie) => PageViewItem(
+                          movie: movie,
+                          pageViewRouteName: NowPlayingMoviesPageView.routeName,
+                        ),
+                      )
+                      .toList(),
+                );
+              case Status.error:
+                return Center(
+                  child: PlainText(
+                    text: '${snapshot.error.toString()}',
+                  ),
+                );
+              case Status.empty:
+                return const Center(
+                  child: PlainText(
+                    text: NowPlayingMoviesPageView.emptyMsg,
+                  ),
+                );
```

Comment:
you are using the same widget for error and empty state, let's do an abstraction

Reply:
done